### ISSUE TITLE
Consider stacked h1 pattern for guides

### ISSUE THREAD

**Issue by willguv** [localgovdrupal/localgov_guides#122]
The NHS guide page equivalent uses a single h1 for titles: page title above, overview title beneath

https://www.nhs.uk/conditions/acute-pancreatitis/causes/

![Image](https://github.com/localgovdrupal/localgov_guides/assets/708140/c66aa0d3-58d7-45a0-9cd7-e2902c2b9a6f)

The content group is enthusiastic about this change

We need to:

- consider as a default for new installs
- provide as an option to existing councils (as they may need to make theming changes to properly implement it)
- discuss whether this should be the default for step by step pages as well (replacing the "part of" panel)

![Image](https://github.com/localgovdrupal/localgov_guides/assets/708140/4094df3f-0fd1-4aa7-87a9-14e98f529dc3)

**Comment by willguv** [comment]
Needs a bit more thought about how to turn on/ off

Challenge is how to roll out and not affect existing sites - not a block, so where does it live? Could be theme option

**Comment by ctorgalson** [comment]
@3mkay  and I had a think about this, and I think we have a workable solution, _assuming_ that the place to solve the problem is in the individual modules that consume the page header block, and not at the module that provides it.

In any case, we wanted to get some feedback before creating a PR, but the reasoning goes like this:

- we currently [set the title with `$event->setTitle()`](https://github.com/localgovdrupal/localgov_guides/blob/2.x/src/EventSubscriber/PageHeaderSubscriber.php#L43),
- `setTitle()` (like `setLede()`) will accept a render array,
- so `localgov_guides` could provide a `hook_theme()` implementation and a corresponding `page-header.html.twig` template,
- then, `setTitle()` in the event subscriber could do basically this:

      $event->setTitle([
        '#theme' => 'page_header',
        '#node' => $overview,
      ]);

To maintain backwards compatibility, the `page-header.html.twig` template provided by the module could basically be

```twig
{{ node.title }}
```

...but since we pass in the node, template overrides could use whatever node data (and markup) they need.

The main question I have is what we'd use to handle the differences between Guide Overviews and Guide pages:

1. logic in the event subscriber, preserving the existing behaviour for Guide Overview pages but permitting customization of Guide Page titles,
2. template suggestions from the module based on bundle,
3. just twig logic based on `node.bundle`,

(1) is probably simplest, (2) is probably most flexible, and (3) is probably most like what's already happening.

---

Does this sound like a workable solution? I'm not very deep into this module, so I could easily be overlooking something important.

**Comment by ctorgalson** [comment]
@willguv Any thoughts on this? We can put together a PR if it seems like a sensible approach to you.

**Comment by willguv** [comment]
Hi @ctorgalson sorry I hadn't replied to this - thanks for the nudge

Yes please go ahead, very kind of you!

Copying @stephen-cox

**Comment by willguv** [comment]
Hi @ctorgalson are you at Annertech by any chance? I think Haringey might have asked for this work already

**Comment by ctorgalson** [comment]
@willguv I am. Their request was what started us looking at this. I should have a chance to work on it soon.

**Comment by willguv** [comment]
Hi @ctorgalson hope all's well. Any news on this? Many thanks, Will

**Comment by andybroomfield** [comment]
See technical discussion and implementation for guides on https://github.com/localgovdrupal/localgov_guides/pull/133 and possible more generic solution for opt in use on any content type on https://github.com/localgovdrupal/localgov_core/pull/193

**Comment by willguv** [comment]
We're following NHS' lead by including both the title and lede in the h1

Agreed in the product group, @andybroomfield present

**Comment by willguv** [comment]
Hi @ctorgalson @markconroy, are you still working on this? Many thanks, Will

**Comment by willguv** [comment]
@markconroy would be good to pick this up and finish it if possible

**Comment by 3mkay** [comment]
@willguv my understanding is that there was a change made upstream during our work which means we need to re-work what we’ve delivered on the HLBC site for it to apply.

**Comment by willguv** [comment]
Thanks @3mkay - is it a big job?

**Comment by andybroomfield** [comment]
Instructions in the 2.13 release notes
https://github.com/localgovdrupal/localgov_core/releases/tag/2.13.0

**Comment by ctorgalson** [comment]
I’m a bit confused. The release notes show a good way for any given LGD implementation to make use of the changes, but I’m unclear what we are proposing to do *here* (also, I think localgov_base might still need to print that `subtitle` variable, but I can’t check at the moment).
  
On Wed, May 22, 2024 at 21:12, Andy Broomfield ***@***.***&gt; wrote:  Instructions in the 2.13 release notes

https://github.com/localgovdrupal/localgov_core/releases/tag/2.13.0

**Comment by markconroy** [comment]
Here's a PR to add the subtitle variable for `localgov_base` - https://github.com/localgovdrupal/localgov_base/pull/561

**Comment by willguv** [comment]
@msayoung I vaguely remember writing this because a council raised accessibility and usability issues with the existing Guides pattern

There was a lot of enthusiasm for the change in the content group

Mark's done so enabling work to make this much easier to implement. Should we go ahead and make this change? If so, should it be a default or an option for councils? 

Thanks

**Comment by willguv** [comment]
@markconroy should we go ahead and implement this change/ give councils an option to use it?

**Comment by markconroy** [comment]
Yes, I think it's okay to merge this one. We chatted about it a lot and gone around the implementation a few times.

**Comment by willguv** [comment]
@markconroy was this ever merged? Thanks

**Comment by markconroy** [comment]
I think this is merged and working, a-la notes for [LocalGov Core 2.13.0](https://github.com/localgovdrupal/localgov_core/releases/tag/2.13.0)

So I _think_ this can be closed. 

@ctorgalson Any reason to keep this issue open?

**Comment by willguv** [comment]
@markconroy did we say this is was something that councils could choose and have to switch on? I think we were cautious about changing the layout - titles may read well placed one on top of another, styling may need local work. What do you think please? Thanks

**Comment by markconroy** [comment]
As far as I remember, it needs some code to be turned on. I think the header block event subscriber needs to be extended for each site to work. @andybroomfield can you confirm?

**Comment by andybroomfield** [comment]
There needs to be an event subscriber for the pageHeader event which can set a subtitle which stacks with the h1.
https://github.com/localgovdrupal/localgov_core/blob/37955444e966652d92a159e25a08118be2908c53/src/Event/PageHeaderDisplayEvent.php#L129C19-L131

**Comment by willguv** [comment]
@markconroy add 'use stacked header on guides' as an localgov_base option, default to off

**Comment by markconroy** [comment]
@willguv I have a PR open now in LocalGov Base to add a toggle for turning this on/off for guides. 

But now I realise there is no "subtitle" field for guides, so I think we now need a chat about

1. Do we need a subtitle field for guides?
2. Do we enable it for new sites only for all sites?
3. What are our thoughts on current sites that may have already implemented a subtitle field and now we are bursting on to the scene with another one?

**Comment by willguv** [comment]
@markconroy see this example

https://www.croydon.gov.uk/adult-health-and-social-care/travel-passes-and-support/blue-badges/fraud

Big black text is 'Blue badge fraud'
Smaller grey text is 'Blue badges'

**Comment by willguv** [comment]
Hi @markconroy any update on this please? Thanks

**Comment by willguv** [comment]
PB 30 Jan 2025 - @markconroy is working on a settings form so councils can choose new pattern site wide

**Comment by markconroy** [comment]
This PR - https://github.com/localgovdrupal/localgov_base/pull/716 - should fix this. It has 2 things in it

1. Stacked header
2. Vertical navigation
